app/src/main/java/net/gsantner/markor/ui/AttachImageOrLinkDialog.java

```diff
@@ -113,6 +118,13 @@ public static Dialog showInsertImageOrLinkDialog(final int action, final int tex
         final FilesystemViewerData.SelectionListener fsListener = new FilesystemViewerData.SelectionListenerAdapter() {
             @Override
             public void onFsViewerSelected(final String request, final File file) {
+
+                if (textFormatId == TextFormat.FORMAT_ZIMWIKI) {
+                    ZimPageFilePathUtil.copyFileToZimPageFolder(file, currentWorkingFile, activity);
+                    inputPathUrl.setText("./"+file.getName());
+                    return;
+                }
```

Comment:
This is the main thing of my dislike. It's free to automatic rewrite the filename, or the write directory...but don't reinvent the file copying stuff and, there should really not be a need here to make something Zim specific.